### PR TITLE
Migrate `src/api/logger-api` and `src/api/propensity-api` TypeScript

### PR DESCRIPTION
**Pull request by ChrisAntaki**
- Migrates to TypeScript
- Updates `src/runtime/event-type-mapping-test.js` test
  - Updates an import
  - Simplifies two tests
  - Removes a redundant test

### CODE REVIEW

**Comment by ChrisAntaki** [comment]
Oops, this isn't quite ready for review, but should be by the end of the day

**Comment by ChrisAntaki** [comment]
It's ready now, I had to improve a test 🧹

File: src/api/logger-api.ts [hunk 1]
```diff
@@ -212,30 +211,23 @@ export const Event = {
...
    * Send a buy-flow event that occurred on the publisher's site to Google.  The
    * ultimate destination is controlled by configuration settings.  Publisher
    * configuration available:
    *   enablePropensity - Sends data to the Propensity to Subscribe ads server.
    *   enableSwgAnalytics - Sends data to Google's analytics server for buy-flow
    *     comparison purposes.
-   * @param {!PublisherEvent} userEvent
    */
-  sendEvent(userEvent) {}
+  sendEvent(_userEvent: PublisherEvent): void;
```
Comment (fikus) [inline src/api/logger-api.ts:230]: Why does _userEvent have an underscore prefix in this interface? Same question for the other interfaces.

Comment (ChrisAntaki) [reply src/api/logger-api.ts:230]: This is the standard way to mark a parameter as unused in TypeScript

Comment (fikus) [reply src/api/logger-api.ts:230]: Interface parameters should not need to be marked unused, there is no implementation so all parameters are unused.

Comment (ChrisAntaki) [reply src/api/logger-api.ts]: Good catch, I just removed the leading `_`s

File: src/api/logger-api.ts [hunk 2]
```diff
@@ -247,9 +239,9 @@ export class LoggerApi {
    *     {'product': ['product1', 'product2']}
    * Each call to this API should have the first argument
    * as a valid string from the enum SubscriptionState.
-   * @param {SubscriptionState} state
-   * @param {?JsonObject} jsonProducts
    */
-  sendSubscriptionState(state, jsonProducts) {}
+  sendSubscriptionState(
+    state: SubscriptionState,
+    jsonProducts: {[key: string]: string[]}
```
Comment (fikus) [inline src/api/logger-api.ts:245]: Can this be typed more precisely with {product: string[]}?

**Review by fikus — APPROVED** [review]

File: src/api/logger-api.ts (hunk 2 above)
Comment (ChrisAntaki) [reply src/api/logger-api.ts]: Updated!

**Pull request merged**